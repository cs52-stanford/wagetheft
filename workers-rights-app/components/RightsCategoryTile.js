import React from "react";
import {
  TouchableOpacity,
  View,
  Text,
  StyleSheet,
  Image,
  Dimensions,
} from "react-native";
import PropTypes from "prop-types";
import Colors from "../constants/Colors";

/*
 * Functional Component Defintion: RightsCategoryTile
 *
 * This component outlines the look of the RightsCategoryTile
 * that is presented on the RightsScreen.
 *
 */

const RightsCategoryTile = ({ onSelect, image, title }) => {
  return (
    <TouchableOpacity style={styles.gridItem} onPress={onSelect} testID="tile">
      <View style={styles.container}>
        <Image style={styles.imageStyle} source={image} />
        <View style={styles.textContainer}>
          <Text testID="text">{title}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const tileWidth = Dimensions.get("window").width * 0.38;
const styles = StyleSheet.create({
  gridItem: {
    width: tileWidth,
    height: tileWidth,
    margin: 10,
    backgroundColor: Colors.lightOrange,
    borderRadius: 15,
    shadowOpacity: 0.25,
    shadowColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 6,
    elevation: 3,
  },
  container: {
    flex: 1,
    padding: 10,
    alignItems: "center",
    justifyContent: "center",
  },
  imageStyle: {
    position: "absolute",
    top: "20%",
    width: 80,
    height: 80,
  },
  textContainer: {
    position: "absolute",
    bottom: "10%",
  },
});

RightsCategoryTile.propTypes = {
  onSelect: PropTypes.func.isRequired,
  image: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};

export default RightsCategoryTile;
