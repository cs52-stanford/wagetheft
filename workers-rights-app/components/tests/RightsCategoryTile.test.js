import React from "react";
import RightsCategoryTile from "../RightsCategoryTile";
import { render, fireEvent } from "@testing-library/react-native";

describe("<RightsCategoryTile /> Tests", () => {
  it("should match snapshot", () => {
    const mockPressFunction = jest.fn();
    const rendered = render(
      <RightsCategoryTile
        title="Test Title"
        onSelect={mockPressFunction}
        image={1}
      />
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it("should display text properly", () => {
    const mockPressFunction = jest.fn();
    const rendered = render(
      <RightsCategoryTile
        title="Test Title"
        onSelect={mockPressFunction}
        image={1}
      />
    );
    const textComponent = rendered.getByTestId("text");
    expect(textComponent.props.children).toEqual("Test Title");
  });

  it("should fire onPress events", () => {
    const onSelect = jest.fn();
    const rendered = render(
      <RightsCategoryTile title="Test Title" onSelect={onSelect} image={1} />
    );
    const buttonComponent = rendered.getByTestId("tile");
    fireEvent(buttonComponent, "press");
    expect(onSelect).toHaveBeenCalled();
  });
});
