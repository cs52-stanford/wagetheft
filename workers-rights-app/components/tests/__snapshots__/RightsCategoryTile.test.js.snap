// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightsCategoryTile /> Tests should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF6EB",
      "borderRadius": 15,
      "elevation": 3,
      "height": 285,
      "margin": 10,
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 6,
      "width": 285,
    }
  }
  testID="tile"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 80,
          "position": "absolute",
          "top": "20%",
          "width": 80,
        }
      }
    />
    <View
      style={
        Object {
          "bottom": "10%",
          "position": "absolute",
        }
      }
    >
      <Text
        testID="text"
      >
        Test Title
      </Text>
    </View>
  </View>
</View>
`;
